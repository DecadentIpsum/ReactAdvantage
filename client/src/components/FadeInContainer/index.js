import React from 'react';
import { CSSTransition } from 'react-transition-group';
import './index.css';

const FadeInContainer = props => (
    <CSSTransition
        classNames="fade"
        {...props}
        timeout={{ enter: 200, exit: 200 }}
    />
);

export default FadeInContainer;
