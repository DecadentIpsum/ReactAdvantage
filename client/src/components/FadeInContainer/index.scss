@import "global-styles/common.scss";

.fade-enter {
    opacity: 0.01;
    transform: translate(0, 20px);
}

.fade-enter.fade-enter-active {
    opacity: 1;
    transform: translate(0, 0);
    transition: opacity 200ms ease-in, transform 200ms ease-in;
}

.fade-exit {
    opacity: 1;
    transform: translate(0, 0);
}

.fade-exit.fade-exit-active {
    opacity: 0.01;
    transform: translate(0, 20px);
    transition: opacity 200ms ease-in, transform 200ms ease-in;
}