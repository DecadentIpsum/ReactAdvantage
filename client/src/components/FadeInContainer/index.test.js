import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import FadeInContainer from './index';

it('renders correctly', () => {
    const component = shallow(
        <FadeInContainer>
            <div>Container content</div>
        </FadeInContainer>
    );

    expect(component).toMatchSnapshot();
});
