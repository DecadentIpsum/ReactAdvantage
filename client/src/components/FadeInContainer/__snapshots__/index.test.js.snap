// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FadeInContainer>
    <div>
        Container content
    </div>
</FadeInContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        Container content
</div>,
      "classNames": "fade",
      "timeout": Object {
        "enter": 200,
        "exit": 200,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Container content",
      },
      "ref": null,
      "rendered": "Container content",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          Container content
</div>,
        "classNames": "fade",
        "timeout": Object {
          "enter": 200,
          "exit": 200,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Container content",
        },
        "ref": null,
        "rendered": "Container content",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
